platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/MarkerFileWriter.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Writes marker files with the given filename to disk in the configured directory.  If there is no configured
+ * directory, no marker files are written.
+ */
+public class MarkerFileWriter {
+
+    /**
+     * The logger for this class.
+     */
+    private static final Logger logger = LogManager.getLogger(MarkerFileWriter.class);
+
+    /**
+     * Rate limited logger for failure to write marker files.
+     */
+    private final RateLimitedLogger failedToWriteMarkerFileLogger;
+
+    /**
+     * The directory where the marker files are written.  If null, no marker files are written.
+     */
+    private final Path markerFileDirectory;
+
+    private final Map<String, Void> markerFilesWrittern = new ConcurrentHashMap<>();
+
+    /**
+     * Creates a new {@link MarkerFileWriter} with the given {@link PlatformContext}.  If the marker file writer is
+     * enabled, files are written to the configured directory.
+     *
+     * @param platformContext the platform context containing configuration.
+     */
+    public MarkerFileWriter(@NonNull final PlatformContext platformContext) {
+        final boolean enabled = platformContext
+                .getConfiguration()
+                .getConfigData(PathsConfig.class)
+                .writePlatformMarkerFiles();
+        final Path markerFileDirectoryPath = platformContext
+                .getConfiguration()
+                .getConfigData(PathsConfig.class)
+                .getMarkerFilesDir();
+        failedToWriteMarkerFileLogger = new RateLimitedLogger(logger, platformContext.getTime(), Duration.ofMinutes(1));
+        Path directory = null;
+        if (enabled) {
+            if (markerFileDirectoryPath != null) {
+                try {
+                    Files.createDirectories(markerFileDirectoryPath);
+                    directory = markerFileDirectoryPath;
+                } catch (final IOException e) {
+                    logger.error(
+                            LogMarker.EXCEPTION.getMarker(),
+                            "Failed to create marker file directory: {}",
+                            markerFileDirectoryPath,
+                            e);
+                }
+            } else {
+                logger.error(LogMarker.STARTUP.getMarker(), "Marker file writing is turned on but directory is null.");
+            }
+        }
+        markerFileDirectory = directory;
+    }
+
+    /**
+     * Writes a marker file with the given filename to the configured directory.
+     *
+     * @param filename the name of the marker file to write.
+     */
+    public void writeMarkerFile(@NonNull final String filename) {
+        if (markerFileDirectory == null) {
+            // Configuration did not set a marker file directory.  No need to write marker file.
+            return;
+        }
+        markerFilesWrittern.computeIfAbsent(filename, key -> {
```

Comment:
If I'm interpreting this doc correctly, the way you're doing it here won't work

```
If the specified key is not already associated with a value (or is mapped to null), attempts to compute its value using the given mapping function and enters it into this map unless null.
```

Since you always return `null` as the computed value, seems like the computation will be made every time, regardless